### ISSUE TITLE
[Pipeline] Extending Stable Diffusion for generating videos

### ISSUE THREAD

**Issue by sayakpaul** [huggingface/diffusers#2432]
Since [Tune-A-Video: One-Shot Tuning of Image Diffusion Models for Text-to-Video Generation](https://arxiv.org/abs/2212.11565) has been there for sometime now, it'd be cool to officially have it supported from Diffusers 🧨 

The best part is the official repository (https://github.com/showlab/Tune-A-Video) itself builds on top of Diffusers.

Architecture-wise, the main change is to inflate the UNet to operate spatiotemporally. This is implemented in the [`UNet3DConditionModel`](https://github.com/showlab/Tune-A-Video/blob/f189b73bab91de7e7ab093dd9bfde77c733d2e6e/tuneavideo/models/unet.py#L37). 

@zhangjiewu will it be possible to publish a few weights on the Hugging Face Hub for the community to try out quickly? Happy to help with the process :) 

We're more than happy to help if a community member wants to pick this up. As it will be the first end-to-end video pipeline in `Diffusers`, I'm very excited about it.

**Comment by Abhinay1997** [comment]
@sayakpaul I’d like to work on this.

**Comment by sayakpaul** [comment]
Please feel free to proceed right away! We are more than happy to help.

**Comment by sayakpaul** [comment]
Seems like there are a few pre-trained models here already: https://huggingface.co/Tune-A-Video-library

**Comment by Abhinay1997** [comment]
@sayakpaul what is the expected outcome ? 

My understanding is:
1. We make the TuneAVideoPipeline available via diffusers.
2. We provide some trained TuneAVideoPipeline compatible checkpoints for users to play around with.
3. We add tests

Let me know if I'm missing something

**Comment by sayakpaul** [comment]
Yes, that is correct. However, for the pipeline to work, we first need to add the inflated classes of UNet and its related components, such as attention.

**Comment by Abhinay1997** [comment]
Ohh Right. Well, I'll start and open a draft PR.

**Comment by jorgemcgomes** [comment]
@Abhinay1997 FYI,  some finding based on my own experiments with Tune-A-Video:

1. using prior preservation loss (as implemented in https://github.com/bryandlee/Tune-A-Video/blob/main/train.py) helps a lot with the relevance of the output videos and in avoiding overfitting. 
This is not part of the official Tune-A-Video implementation, but it's something you might want to consider when building the training script.

2. Using more training frames (I tried up to 32) resulted in a more stable loss and higher quality output. So, definitely don't hardcode it to the 8 frames found in the official implementation.

**Comment by sayakpaul** [comment]
Thank you for sharing your experience, definitely very helpful to keep in mind. However, for the first PR, we won't be considering the training script.

**Comment by jorgemcgomes** [comment]
@sayakpaul Yes, definitely. I'll keep an eye on the PR that @Abhinay1997 will open 👍

**Comment by Abhinay1997** [comment]
@jorgemcgomes thanks for the inputs. Will keep this in mind. I'm sure we'll need these details down the line.

**Comment by Abhinay1997** [comment]
Hi @sayakpaul added a draft PR #2455.

So, a few follow up questions:-

1. We'll be replacing `einops.rearrange`, `einops.repeat` with equivalent torch operations where applicable, correct ? I see einops being commented out in other files too. Just wanted to confirm.

2. The `BasicTransformerBlock` in TuneAVideo's `attention.py` uses `SparseCausalAttention`. But diffusers already has a `BasicTransformerBlock` with `CrossAttention`.  Can we make the `BasicTransfomerBlock` configurable then ? For now I am calling it `BasicSparseTransformerBlock` to differentiate.

**Comment by sayakpaul** [comment]
I replied on your PR :)

**Comment by zhangjiewu** [comment]
Hi folks, thank you for your great efforts in integrating Tune-A-Video to diffusers. We have made some updates to [our implementation](https://github.com/showlab/Tune-A-Video), resulting in improved consistency. We hope that these changes would be helpful. 

Please let me know if there is anything I can assist you with. :)

**Comment by Abhinay1997** [comment]
Thanks for letting me know @zhangjiewu. I'll cross check to see if I have missed anything. :)

**Comment by Abhinay1997** [comment]
@sayakpaul , @zhangjiewu I ran into an issue while testing the ported pipeline. I'm able to generate the output but the result is noise.

Any advice for debugging ? I guess I made a mistake while porting the `SparseCausalAttention` module. Will cross verify that.

Warning ! Flashing image
https://user-images.githubusercontent.com/24771261/222959721-dda8bc27-02fe-4f72-88a5-53ee68200056.gif

**Comment by tune-a-video** [comment]
hi @Abhinay1997, have you been able to solve the problem? If not, could you share your code with me so that I can assist you further?

**Comment by zhangjiewu** [comment]
hi @Abhinay1997, have you been able to solve the problem? If not, could you share your code with me so that I can assist you further?

**Comment by Abhinay1997** [comment]
@zhangjiewu didn't get a chance :( Can you please have a quick glance at this :https://github.com/Abhinay1997/diffusers/blob/tune_a_video_port/src/diffusers/models/cross_attention.py#L673

For now, I was just trying to get it to work. Note that I have replaced einops with torch equivalents.

**Comment by Abhinay1997** [comment]
@zhangjiewu never mind. It was a silly mistake. Fixed it. Here's a sample with the ported code:

![ironman_surf](https://user-images.githubusercontent.com/24771261/226119099-5c2a86a1-6783-434f-8be3-3928ebffee61.gif)
